x/supply/keeper/keeper.go

```diff
@@ -0,0 +1,79 @@
+package keeper
+
+import (
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/supply/types"
+)
+
+// Keeper of the supply store
+type Keeper struct {
+	cdc       *codec.Codec
+	storeKey  sdk.StoreKey
+	ak        types.AccountKeeper
+	bk        types.BankKeeper
+	permAddrs map[string]permAddr
+}
+
+type permAddr struct {
+	permission string // holder/minter/burner
+	address    sdk.AccAddress
+}
+
+// NewpermAddr creates a new permAddr object
+func newPermAddr(permission, name string) permAddr {
+	return permAddr{
+		permission: permission,
+		address:    types.NewModuleAddress(name),
+	}
+}
+
+// NewKeeper creates a new Keeper instance
+func NewKeeper(cdc *codec.Codec, key sdk.StoreKey, ak types.AccountKeeper, bk types.BankKeeper,
+	codespace sdk.CodespaceType, holders, minters, burners []string) Keeper {
+
+	// set the addresses
+	permAddrs := make(map[string]permAddr)
+	for _, name := range holders {
```

Comment:
from a simple grok, I don't see where `Holder` is actually used/checked against.

Reply:
renamed to Basic, as it's more intuitive